models.py

```diff
@@ -584,8 +587,22 @@ def forward(self, signal: shaping.BT, mask: shaping.BT = None, **kwargs) -> shap
 
 		power_spectrum = real_squared + imag_squared
 		log_mel_features = self.mel(power_spectrum).log()
+
+		if self.check_features_divisibility_by_32:
+			if signal.shape[-1] % (32 / 2) != 0:
+				warnings.warn('Shape of input signal not divisible by 16')
```

Comment:
please use  `assert condition, 'message'` like we're doing elsewhere. assert is sufficient for our usecase